lib/std/math/fabs.zig

```diff
@@ -16,86 +15,28 @@ const maxInt = std.math.maxInt;
 ///  - fabs(nan)   = nan
 pub fn fabs(x: anytype) @TypeOf(x) {
     const T = @TypeOf(x);
-    return switch (T) {
-        f16 => fabs16(x),
-        f32 => fabs32(x),
-        f64 => fabs64(x),
-        f128 => fabs128(x),
-        else => @compileError("fabs not implemented for " ++ @typeName(T)),
-    };
-}
-
-fn fabs16(x: f16) f16 {
-    var u = @bitCast(u16, x);
-    u &= 0x7FFF;
-    return @bitCast(f16, u);
-}
-
-fn fabs32(x: f32) f32 {
-    var u = @bitCast(u32, x);
-    u &= 0x7FFFFFFF;
-    return @bitCast(f32, u);
-}
-
-fn fabs64(x: f64) f64 {
-    var u = @bitCast(u64, x);
-    u &= maxInt(u64) >> 1;
-    return @bitCast(f64, u);
-}
+    const IntT = std.meta.Int(.unsigned, @bitSizeOf(T));
+    if (@typeInfo(T) != .Float) {
+        @compileError("fabs not implemented for " ++ @typeName(T));
+    }
 
-fn fabs128(x: f128) f128 {
-    var u = @bitCast(u128, x);
-    u &= maxInt(u128) >> 1;
-    return @bitCast(f128, u);
+    const remove_sign = @as(IntT, math.maxInt(IntT)) >> 1;
+    return @bitCast(T, @bitCast(IntT, x) & remove_sign);
```

Comment:
```zig
const uintT = @bitCast(IntT, x);
return @bitCast(T, uintT & remove_sign);
```
is more readable as mixing 2 comptime executions with 1 runtime `&` in a nested scope is not less trivial to read.

Reply:
Done on both fabs and isNormal.